metrics/web/lib/dashboard/presentation/widgets/build_result_bar_component.dart

```diff
@@ -0,0 +1,150 @@
+    const indicatorRadius = indicatorDiameter / 2.0;
+    const indicatorHorizontalOffset = (barWidth - indicatorDiameter) / 2.0;
+
+    if (buildResult == null) {
+      final inactiveTheme = MetricsTheme.of(context).inactiveWidgetTheme;
+
+      return PlaceholderBar(
+        width: barWidth,
+        height: 4.0,
+        color: inactiveTheme.primaryColor,
+      );
+    }
+
+    return BasePopup(
+      popupOpaque: false,
+      closeOnTapOutside: false,
+      popupConstraints: const BoxConstraints(
+        minWidth: _popupWidth,
+        maxWidth: _popupWidth,
+      ),
+      offsetBuilder: (triggerSize) => _calculatePopupOffset(
+        triggerSize,
+        indicatorRadius,
+      ),
+      triggerBuilder: (context, openPopup, closePopup, isOpened) {
+        return MouseRegion(
+          cursor: SystemMouseCursors.click,
+          onEnter: (_) => openPopup(),
+          onExit: (_) => closePopup(),
+          child: GestureDetector(
+            behavior: HitTestBehavior.opaque,
+            onTap: _onTap,
+            child: Padding(
+              padding: _barPadding,
+              child: Stack(
+                clipBehavior: Clip.none,
+                fit: StackFit.passthrough,
+                children: [
+                  BuildResultBar(
+                    buildResult: buildResult,
+                  ),
+                  if (isOpened)
+                    Positioned(
+                      right: indicatorHorizontalOffset,
+                      bottom: -indicatorRadius,
+                      child: _graphIndicator,
+                    ),
+                ],
+              ),
+            ),
+          ),
+        );
+      },
+      popup: BuildResultPopupCard(
+        buildResultPopupViewModel: buildResult.buildResultPopupViewModel,
+      ),
+    );
+  }
+
+  /// A [GraphIndicator] widget to use based on the
+  /// [BuildResultViewModel.buildStatus].
+  Widget get _graphIndicator {
+    switch (buildResult?.buildStatus) {
+      case BuildStatus.successful:
+        return const PositiveGraphIndicator();
+      case BuildStatus.failed:
+        return const NegativeGraphIndicator();
+      case BuildStatus.unknown:
+        return const NeutralGraphIndicator();
+      case BuildStatus.inProgress:
+        return const NeutralGraphIndicator();
+    }
+
+    return null;
+  }
+
+  /// An [EdgeInsets] to apply to this bar component.
+  EdgeInsets get _barPadding {
+    return paddingStrategy.getBarPadding(buildResult);
+  }
+
+  /// Calculates the [Offset] for the bar popup.
+  Offset _calculatePopupOffset(
+    Size triggerSize,
+    double indicatorRadius,
+  ) {
+    final paddingOffset = _barPadding.left - _barPadding.right;
+    final dx = triggerSize.width / 2 - _popupWidth / 2 + paddingOffset / 2;
+    final dy = triggerSize.height + indicatorRadius + 4.0;
+
+    return Offset(dx, dy);
+  }
+
+  /// Opens the [BuildResultViewModel.url].
+  Future<void> _onTap() async {
```

Comment:
Would it be great to give this method a bit more descriptive name? Maybe something like `openBuildUrl` or `launchBuildUrl`?